reccmp/isledecomp/compare/asm/parse.py

```diff
@@ -58,40 +59,38 @@ def __init__(
     def reset(self):
         self.replacements = {}
 
-    def is_relocated(self, addr: int) -> bool:
-        if callable(self.relocate_lookup):
-            return self.relocate_lookup(addr)
+    def is_addr(self, value: int) -> bool:
+        """Wrapper for user-provided address test"""
+        if callable(self.addr_test):
+            return self.addr_test(value)
 
         return False
 
-    def lookup(
-        self, addr: int, use_cache: bool = True, exact: bool = False
-    ) -> Optional[str]:
-        """Return a replacement name for this address if we find one."""
-        if use_cache and addr in self.replacements:
-            return self.replacements[addr]
-
+    def lookup(self, addr: int, exact: bool = False) -> Optional[str]:
+        """Wrapper for user-provided name lookup"""
         if callable(self.name_lookup):
-            if (name := self.name_lookup(addr, exact)) is not None:
-                if use_cache:
-                    self.replacements[addr] = name
-
-                return name
+            return self.name_lookup(addr, exact=exact)
 
         return None
 
-    def replace(self, addr: int) -> str:
-        """Same function as lookup above, but here we return a placeholder
-        if there is no better name to use."""
-        if (name := self.lookup(addr)) is not None:
+    def replace(self, value: int, exact: bool = False, test: bool = False) -> str:
+        """Provide a replacement name for the given address."""
+        if test and not self.is_addr(value):
+            return hex(value)  # TODO.
```

Comment:
What is the status of this TODO?

Reply:
Right now we convert the hex string to a number (using `int(value, 16)`) and then call `lookup` or `replace`. I was thinking about changing this so that these functions accept the string value instead. This will have to be the case with 16-bit asm because we need to split the `segment:offset` pair for `lcall` instructions and then convert to some normalized value.

I'm not totally committed to this yet, but `replace()` has to return something if the address test fails. The compromise is to do the reverse conversion here. This is sort of wasteful and depends on python `hex` and the `capstone` library having compatible output.

I can try and address this now before merging. I think the tests cover this okay (what you get in is what you get back) but I'm becoming less and less happy with this solution as I continue to type out this comment 😀

Reply:
I'd either fix it or add a link to the comment you just posted to the TODO comment. Personally, I don't like TODO comments without further explanations on `master` since there's always a chance they stay there for a long time and don't make sense to future contributors.